### ISSUE TITLE
Reloading of proxy seems to result in HAProxy zombies over time

### ISSUE THREAD

**Issue by roelio** [vfarcic/docker-flow-proxy#245]
```ps aux | grep haproxy
roel      9979  0.0  0.0  14196  1016 pts/0    S+   20:51   0:00 grep --color=auto haproxy
root     20846  0.0  0.0      0     0 ?        Zs   20:38   0:00 [haproxy] <defunct>
root     20972  0.0  0.0  14164   808 ?        Ss   20:38   0:00 haproxy -f /cfg/haproxy.cfg -D -p /var/run/haproxy.pid -sf 11```

**Comment by rics3n** [comment]
I experienced this problem as well. Both on docker for mac in swarm mode and on an ubuntu cluster.  A while back this was a bug in consul-template as well. It originates from the usage of the -sf instead of -st option from haproxy and reaping processes in docker. I got it working with hashicorps reap lib. Here is some more detail about reaping https://github.com/hashicorp/consul-template/issues/428

**Comment by morph027** [comment]
tini might be worth a look too: https://github.com/krallin/tini

Been using it for quite some time.

**Comment by vfarcic** [comment]
I did not manage to add **tini** to the project. I gave up to fast so that does not mean that it is not doable. @morph027 would you like to give it a try with a PR?

**Comment by morph027** [comment]
Sure, just need some time to reproduce the issue and then see if it's solved afterwards :+1:

**Comment by vfarcic** [comment]
The issue is fixed and available in the release `17.09.14-15`.

Thanks @carlhals for the PR #328.
